control-plane/api-gateway/cache/consul.go

```diff
@@ -222,7 +223,7 @@ func (c *Cache) updateAndNotify(ctx context.Context, once *sync.Once, kind strin
 
 	for _, entry := range entries {
 		meta := entry.GetMeta()
-		if meta[constants.MetaKeyKubeName] == "" || meta[constants.MetaKeyDatacenter] != c.datacenter {
+		if (meta[constants.MetaKeyKubeName] == "" || meta[constants.MetaKeyDatacenter] != c.datacenter) && kind != api.JWTProvider {
```

Comment:
I kinda feel icky about this but I'm not sure of another way to skip this if we get a JWTProvider back

Reply:
Agree. I think nesting the `if` might make this easier to comprehend?